oasislmf/preparation/gul_inputs.py

```diff
@@ -333,7 +333,7 @@ def get_gul_input_items(
 
     # this block gets fired if the hash_group_ids is True
     else:
-        gul_inputs_df["group_id"] = (pd.util.hash_pandas_object(gul_inputs_df[group_id_cols]).to_numpy() >> 33)
+        gul_inputs_df["group_id"] = pd.util.hash_pandas_object(gul_inputs_df[group_id_cols])
```

Comment:
I agree that by introducing `drop=True` in `reset_index()` here we know that indices are not duplicate, but in any case we need to ensure we compute hashes only using the desired columns:
```suggestion
        gul_inputs_df["group_id"] = pd.util.hash_pandas_object(gul_inputs_df[group_id_cols], index=False)
```

Reply:
I can't comment on the bitshift `>> 33`. I don't understand why we needed it in the first place, so please @sstruzik comment on this. In my view, we might not need to divide by 2**33 anymore, but please confirm.

Reply:
@sstruzik suggested that we keep the >> 33 because  "The bit shift is to have value in the int32 range"